### PR TITLE
misshapen scopus citations are now quietly dropped.

### PR DESCRIPTION
**Pull request by lsh-0**
failures to parse bad data are still written to disk for later.

this is an attempt to remove known bad results from scopus search results as bad entries are written to disk and recorded as an error which is then sent to us daily in a log alert.

### CODE REVIEW

File: src/metrics/scopus/citations.py [hunk 1]
```diff
@@ -82,16 +82,24 @@ def search(api_key=settings.SCOPUS_KEY, doi_prefix=settings.DOI_PREFIX):
...
+        ensure(entry['prism:doi'].startswith(settings.DOI_PREFIX), 'entry has an unknown DOI prefix: %r' % entry['prism:doi'])
+        return {
+            'doi': entry['prism:doi'],
+            'num': int(entry['citedby-count']),
+            'source': models.SCOPUS,
+            'source_id': citedby_link['@href']
+        }
+    except ParseError:
+        # these won't be caught by handler.capture_parse_error
+        LOG.warn("discarding scopus citation: failed to parse response", extra={'reponse': entry})
```
Comment (giorgiosironi) [inline src/metrics/scopus/citations.py]: `reponse` a typo?

File: src/metrics/tests/test_scopus_citations.py [hunk 2]
```diff
@@ -25,6 +26,27 @@ def test_scopus_parse_entry(self):
...
+
+    def test_scopus_unparseable_entry(self):
+        "unparseable entries have a different return type"
+        with patch('metrics.handler.writefile') as mock:
+            entry = self.good_entry
+            # skitch entry so it's bad
+            del entry['prism:doi']
+            result = citations.parse_entry(entry)
+            self.assertTrue('bad' in result) # ll: {'bad': ...}
+            # a bad entry is still returned but the body isn't written to disk for inspection
```
Comment (giorgiosironi) [inline src/metrics/tests/test_scopus_citations.py]: I wonder why we don't store anything for this case?

File: src/metrics/scopus/citations.py (hunk 1 above)
Comment (lsh-0) [reply src/metrics/scopus/citations.py]: nice catch

File: src/metrics/tests/test_scopus_citations.py (hunk 2 above)
Comment (lsh-0) [reply src/metrics/tests/test_scopus_citations.py]: if the record is missing fields, it's hard to create the minimum viable Article object. Scopus is returning about 23 of these I think, so 23 errors every day we can't recover from. In this case, I'm plain ignoring them. Well, a warning is issued, but that's about it. 

If those fields are present though and have bad values, there is room for wrangling, like the '10.7554/elife.e123456' formatted doi I came across the other day.

**Pull request merged**